src/modules/daemonconfiguration/src/common.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved..
+// Licensed under the MIT License.
+
+use libc::{c_int, EINVAL, ENOMEM};
+use std::ffi::NulError;
+use std::fmt;
+use std::str::Utf8Error;
+
+#[derive(Debug, PartialEq)]
+pub enum MmiError {
+    FailedRead,
+    FailedAllocate,
+    InvalidArgument,
```

Comment:
Consider adding context for these errors. Ex: `InvalidArgument(String)` to provide the name of the argument in log traces.

Reply:
We'd discussed this for the Sample module! More specific logging is left at the rust lib level (specific logging can be done before the error is returned). But it's hard to use the specific String arg with a wide catching match statement because the String can't be incorporated into the fmt::display return string